librz/core/cmd/cmd_print.c

```diff
@@ -4483,64 +4483,60 @@ static inline int cmd_pxb_k(const ut8 *buffer, int x) {
 	return buffer[3 - x] << (8 * x);
 }
 
-static void print_json_string(RzCore *core, const char *block, ut32 len, RzCoreStringKind kind) {
-	RzCoreString *cstring = rz_core_string_information(core, block, len, kind);
-	if (!cstring) {
-		return;
+static inline char *get_section_name(RzCore *core) {
+	const char *csection = rz_core_get_section_name(core, core->offset);
+	if (RZ_STR_ISEMPTY(csection)) {
+		return strdup("unknown");
 	}
-	// cleaning useless spaces in section name in json data.
-	char *section_name = strdup(rz_str_trim_head_ro(cstring->section_name));
-	char *p;
-	for (p = (char *)section_name; *p && *p != ' '; p++) {
+	csection = rz_str_trim_head_ro(csection);
+	char *section_name = strdup(csection);
+	rz_str_trim_tail(section_name);
+	if (RZ_STR_ISEMPTY(section_name)) {
+		free(section_name);
+		return strdup("unknown");
 	}
-	*p = '\0';
-	char *kindstr;
-	switch (cstring->kind) {
-	case RZ_CORE_STRING_KIND_ASCII:
-		kindstr = "ascii";
-		break;
-	case RZ_CORE_STRING_KIND_WIDE16: // No UTF/Unicode encoding
-		kindstr = "wide16";
-		break;
-	case RZ_CORE_STRING_KIND_WIDE32: // NO UTF/Unicode encoding
-		kindstr = "wide32";
-		break;
-	case RZ_CORE_STRING_KIND_UTF8:
-		kindstr = "utf8";
-		break;
-	case RZ_CORE_STRING_KIND_UTF16:
-		kindstr = "utf16";
-		break;
-	case RZ_CORE_STRING_KIND_UTF32:
-		kindstr = "utf32";
-		break;
-	default:
-		kindstr = "unknown";
-		break;
+	return section_name;
+}
+
+static void print_json_string(RzCore *core, const ut8 *block, ut32 len, RzStrEnc encoding, bool stop_at_nil) {
+	const char *kindstr = rz_str_enc_as_string(encoding);
```

Comment:
could be moved later in the code

Reply:
where you want me to move this?

Reply:
After the next check, so the unnecessary execution of that line could be avoided. Miniscule but free optimization ;)